### PR TITLE
[SITES-425] - Correctly determine if a node with a given name exists

### PR DESCRIPTION
**Pull request by tessereth**
Note: using find_by for name on a Node object actually queries the name column rather than `content[:name]`.

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -16,37 +16,30 @@
 topic.summary = 'The business section covers a range of business-related topics.'
 topic.save
 
-news1 = NewsArticle.find_or_create_by!({
- name: "Business News",
+news1 = NewsArticle.with_content_value(:name, "Business News")
+            .find_or_create_by!({
```
Comment (ltankey) [inline db/seeds.rb]: This indenting is a bit weird compared to the lines below... /pedant.

**Comment by ltankey** [comment]
LGTM

File: app/models/node.rb [hunk 2]
```diff
@@ -14,6 +14,23 @@ class Node < ApplicationRecord
 
   scope :published, -> { where state: 'published' }
 
+  # These methods aren't DRY to avoid inserting
+  # unescaped strings directly into a PG query.
+  #
+  # This type of query requires the store column to be
+  # unescaped.
+  scope :with_content_value, -> (field, value) {
```
Comment (tobyhede) [inline app/models/node.rb]: Awesome. I love PG.

**Comment by tobyhede** [comment]
LGTM

**Pull request merged**